### ISSUE TITLE
[Metal][Clang + DXC] Buffer robustness failure in 64 bit Interlocked* tests

### ISSUE THREAD

**Issue by bob80905** [llvm-beanz/HLSLTest#1164]
In the link below the error is seen:
```
project/build/tools/OffloadTest/test/mtl/Feature/HLSLLib/Output/InterlockedAdd.int64.test.tmp.o
Validation Warning: [ mvk-warn ]
VK_ERROR_FEATURE_NOT_PRESENT: Metal does not support buffer robustness.

gpu-exec: error: Compilation failed due to an interrupted connection: XPC_ERROR_CONNECTION_INTERRUPTED. This error occurred after multiple retries.: 

--

********************
```
https://github.com/llvm/offload-test-suite/actions/runs/25528342754/job/74928931859?pr=1163#step:13:117

This was seen on check-hlsl-mtl.

**Comment by bogner** [comment]
I think the title here is pointing at the spurious moltenvk warning and not the actual issue we're hitting. I see a slightly more helpful message locally and on [more recent runs](https://github.com/llvm/offload-test-suite/actions/runs/33932005629/job/101212451334?pr=1500#step:15:143):
> gpu-exec: error: Failed to compile and link DXIL to Metal IR: IRErrorCodeUnsupportedInstruction

Looks to me like `@dx.op.atomicBinOp.i64` isn't handled by the metal shaderconverter.
